src/gar.lua

```diff
@@ -949,61 +977,23 @@ function gar.instantDelegateWithdrawal(from, gatewayAddress, vaultId, currentTim
 		processInstantWithdrawal(vault.balance, elapsedTime, totalWithdrawalTime, from)
 
 	-- Remove the vault after withdrawal
-	delegate.vaults[vaultId] = nil
-
-	-- Remove the delegate if no stake is left
-	if delegate.delegatedStake == 0 and next(delegate.vaults) == nil then
-		gar.pruneDelegateFromGateway(from, gateway)
+	if isGatewayWithdrawal then
+		gateway.vaults[vaultId] = nil
+	else
+		if delegate == nil then
+			error("Delegate not found")
+		end
+		delegate.vaults[vaultId] = nil
```

Comment:
This needs to use `gar.pruneDelegateFromGateway(from, gateway)` to jive with allowlist

Reply:
9e315d0